lib/winrm/http/transport.rb

```diff
@@ -209,6 +211,21 @@ def winrm_decrypt(str)
         end
       end
 
+      def issue_challenge_response(negotiate)
```

Comment:
This is more clear than how we recursively called `send_request` before.

Reply:
Cool - glad that there wasn't a compelling reason the code was written the way it was before. These handshakes can get a little tricky.